stdlib/public/SDK/Foundation/NSStringAPI.swift

```diff
@@ -230,849 +170,1202 @@ extension String {
+
+  /// An uppercase version of the string that is produced using the current
+  /// locale.
+  @available(OSX 10.11, iOS 9.0, *)
+  public var localizedUppercase: String {
+    return _ns.localizedUppercase as String
+  }
+
+  // - (NSString *)uppercaseStringWithLocale:(Locale *)locale
+
+  /// Returns a version of the string with all letters
+  /// converted to uppercase, taking into account the specified
+  /// locale.
+  public func uppercased(with locale: Locale?) -> String {
+    return _ns.uppercased(with: locale)
+  }
+
+  //===--- Omitted due to redundancy with "utf8" property -----------------===//
+  // - (const char *)UTF8String
+
+  // - (BOOL)
+  //     writeToFile:(NSString *)path
+  //     atomically:(BOOL)useAuxiliaryFile
+  //     encoding:(NSStringEncoding)enc
+  //     error:(NSError **)error
+
+  /// Writes the contents of the `String` to a file at a given
+  /// path using a given encoding.
+  public func write(
+    toFile path: String, atomically useAuxiliaryFile:Bool,
+    encoding enc: String.Encoding
+  ) throws {
+    try _ns.write(
+      toFile: path, atomically: useAuxiliaryFile, encoding: enc.rawValue)
+  }
+
+  // - (BOOL)
+  //     writeToURL:(NSURL *)url
+  //     atomically:(BOOL)useAuxiliaryFile
+  //     encoding:(NSStringEncoding)enc
+  //     error:(NSError **)error
+
+  /// Writes the contents of the `String` to the URL specified
+  /// by url using the specified encoding.
+  public func write(
+    to url: URL, atomically useAuxiliaryFile: Bool,
+    encoding enc: String.Encoding
+  ) throws {
+    try _ns.write(
+      to: url, atomically: useAuxiliaryFile, encoding: enc.rawValue)
+  }
+
+  // - (nullable NSString *)stringByApplyingTransform:(NSString *)transform reverse:(BOOL)reverse NS_AVAILABLE(10_11, 9_0);
+
+  /// Perform string transliteration.
+  @available(OSX 10.11, iOS 9.0, *)
+  public func applyingTransform(
+    _ transform: StringTransform, reverse: Bool
+  ) -> String? {
+    return _ns.applyingTransform(transform, reverse: reverse)
+  }
+
+  // - (void)
+  //     enumerateLinguisticTagsInRange:(NSRange)range
+  //     scheme:(NSString *)tagScheme
+  //     options:(LinguisticTaggerOptions)opts
+  //     orthography:(Orthography *)orthography
+  //     usingBlock:(
+  //       void (^)(
+  //         NSString *tag, NSRange tokenRange,
+  //         NSRange sentenceRange, BOOL *stop)
+  //       )block
+
+  /// Performs linguistic analysis on the specified string by
+  /// enumerating the specific range of the string, providing the
+  /// Block with the located tags.
+  public func enumerateLinguisticTags(
+    in range: Range<Index>,
+    scheme tagScheme: String,
+    options opts: NSLinguisticTagger.Options = [],
+    orthography: NSOrthography? = nil,
+    invoking body:
+      (String, Range<Index>, Range<Index>, inout Bool) -> Void
+  ) {
+    _ns.enumerateLinguisticTags(
+      in: _toNSRange(range),
+      scheme: tagScheme,
+      options: opts,
+      orthogr
```

Comment:
it is a bit of a shame that enumerateSubstrings enumerates Strings and not Substrings...